Sources/IO/Geometry/GLTFReader/Parser.js

```diff
@@ -0,0 +1,453 @@
+  }
+
+  resolveSampler(sampler, index) {
+    sampler.id = sampler.id || `sampler-${index}`;
+
+    if (!Object.hasOwn(sampler, 'wrapS')) sampler.wrapS = GL_SAMPLER.REPEAT;
+    if (!Object.hasOwn(sampler, 'wrapT')) sampler.wrapT = GL_SAMPLER.REPEAT;
+
+    if (!Object.hasOwn(sampler, 'minFilter'))
+      sampler.minFilter = GL_SAMPLER.LINEAR_MIPMAP_LINEAR;
+    if (!Object.hasOwn(sampler, 'magFilter'))
+      sampler.magFilter = GL_SAMPLER.NEAREST;
+
+    // Map textual parameters to GL parameter values
+    sampler.parameters = {};
+    for (const key in sampler) {
+      const glEnum = getGLEnumFromSamplerParameter(key);
+      if (glEnum !== undefined) {
+        sampler.parameters[glEnum] = sampler[key];
+      }
+    }
+    return sampler;
+  }
+
+  resolveImage(image, index) {
+    image.id = image.id || `image-${index}`;
+    if (image.bufferView !== undefined) {
+      image.bufferView = this.get('bufferViews', image.bufferView);
+    }
+    return image;
+  }
+
+  resolveBufferView(bufferView, index) {
+    bufferView.id = bufferView.id || `bufferView-${index}`;
+    const bufferIndex = bufferView.buffer;
+    bufferView.buffer = this.buffers[bufferIndex];
+
+    const arrayBuffer = this.buffers[bufferIndex].arrayBuffer;
+    let byteOffset = this.buffers[bufferIndex].byteOffset || 0;
+
+    if ('byteOffset' in bufferView) {
+      byteOffset += bufferView.byteOffset;
+    }
+
+    bufferView.data = new Uint8Array(
+      arrayBuffer,
+      byteOffset,
+      bufferView.byteLength
+    );
+    return bufferView;
+  }
+
+  resolveCamera(camera, index) {
+    camera.id = camera.id || `camera-${index}`;
+    return camera;
+  }
+
+  setupAnimations() {
+    const animations = [];
+    this.json.animations?.forEach((glTFAnimation) => {
+      const animation = new AnimationClip(glTFAnimation.name);
+      glTFAnimation.channels.forEach((channel) => {
+        const sampler = glTFAnimation.samplers[channel.sampler];
+        const target = this.json.nodes[channel.target.node];
+        const path = channel.target.path;
+
+        const times = this.json.accessors[sampler.input].value;
+        const values = this.json.accessors[sampler.output].value;
+        const interpolation = sampler.interpolation || 'LINEAR';
+
+        const track = new AnimationTrack(
+          target,
+          path,
+          times,
+          values,
+          interpolation
+        );
+        animation.addTrack(track);
+      });
+
+      animations.push(animation);
+    });
+
+    return animations;
+  }
+
+  loadBuffers() {
+    const promises = [];
+    return new Promise((resolve, reject) => {
+      this.json.buffers.forEach((buffer, idx) => {
+        promises.push(
+          Promise.resolve(
+            this.loadBuffer(buffer, idx).then(() => {
+              console.log('Buffer loaded');
+              delete buffer.uri;
+            })
+          )
+        );
+      });
+
+      Promise.all(promises).then(() => resolve(this.buffers));
+    });
```

Comment:
If `loadBuffer()` returns a promise, you just need `loadBuffers` to return a unique promise (you do not seem to need to do `resolve(this.buffers)`), you could then simply do: 
```
loadBuffers() {
  const promises = this.json.buffers.map((buffer, idx) => {
    return this.loadBuffer(buffer, idx).then(() => {
      delete buffer.uri;
  });
  return Promise.all(promises);
}
```